Begin generating RemoteLayerTree serialization code

achristensen07 opened the pull request:
#### ec6beab1829ef5be2644a2b8cf61f0792f9e906e
<pre>
Begin generating RemoteLayerTree serialization code
<a href="https://bugs.webkit.org/show_bug.cgi?id=250930">https://bugs.webkit.org/show_bug.cgi?id=250930</a>
rdar://104505653

Reviewed by Simon Fraser.

* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.h:
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(typenames):
(remove_template_parameters):
(remove_alias_struct_or_class):
(alias_struct_or_class):
(generate_header):
(generate_serialized_type_info):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
Include some test expectation that should&apos;ve been included in 259112@main
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.mm:
(WebKit::RemoteLayerBackingStoreCollection::prepareBackingStoresForDisplay):
(WebKit::RemoteLayerBackingStoreCollection::didFlushLayers):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in: Added.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm:
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToLayer):
(WebKit::RemoteLayerTreePropertyApplier::applyProperties):
(WebKit::RemoteLayerTreePropertyApplier::applyHierarchyUpdates):
(WebKit::RemoteLayerTreePropertyApplier::updateMask):
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToUIView):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(WebKit::RemoteLayerTreeTransaction::LayerProperties::encode const):
(WebKit::RemoteLayerTreeTransaction::LayerProperties::decode):
(WebKit::dumpChangedLayers):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStoreCollection.mm:
(WebKit::RemoteLayerWithRemoteRenderingBackingStoreCollection::prepareBackingStoresForDisplay):
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.cpp:
(ArgumentCoder&lt;RequestedScrollData&gt;::encode): Deleted.
(ArgumentCoder&lt;RequestedScrollData&gt;::decode): Deleted.
(ArgumentCoder&lt;SnapOffset&lt;float&gt;&gt;::encode): Deleted.
(ArgumentCoder&lt;SnapOffset&lt;float&gt;&gt;::decode): Deleted.
(ArgumentCoder&lt;FloatScrollSnapOffsetsInfo&gt;::encode): Deleted.
(ArgumentCoder&lt;FloatScrollSnapOffsetsInfo&gt;::decode): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingUIState.cpp:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingUIState.h:
(WebKit::RemoteScrollingUIState::changes const):
* Source/WebKit/Shared/RemoteLayerTree/SwapBuffersDisplayRequirement.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::updateLayerTree):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.cpp:
(WebKit::PlatformCALayerRemote::PlatformCALayerRemote):
(WebKit::PlatformCALayerRemote::recursiveBuildTransaction):
(WebKit::PlatformCALayerRemote::copyContentsFromLayer):
(WebKit::PlatformCALayerRemote::removeSublayer):
(WebKit::PlatformCALayerRemote::setSublayers):
(WebKit::PlatformCALayerRemote::removeAllSublayers):
(WebKit::PlatformCALayerRemote::appendSublayer):
(WebKit::PlatformCALayerRemote::insertSublayer):
(WebKit::PlatformCALayerRemote::replaceSublayer):
(WebKit::PlatformCALayerRemote::addAnimationForKey):
(WebKit::PlatformCALayerRemote::removeAnimationForKey):
(WebKit::PlatformCALayerRemote::setMask):
(WebKit::PlatformCALayerRemote::setClonedLayer):
(WebKit::PlatformCALayerRemote::setOpaque):
(WebKit::PlatformCALayerRemote::setBounds):
(WebKit::PlatformCALayerRemote::setPosition):
(WebKit::PlatformCALayerRemote::setAnchorPoint):
(WebKit::PlatformCALayerRemote::setTransform):
(WebKit::PlatformCALayerRemote::setSublayerTransform):
(WebKit::PlatformCALayerRemote::setHidden):
(WebKit::PlatformCALayerRemote::setContentsHidden):
(WebKit::PlatformCALayerRemote::setUserInteractionEnabled):
(WebKit::PlatformCALayerRemote::setBackingStoreAttached):
(WebKit::PlatformCALayerRemote::setGeometryFlipped):
(WebKit::PlatformCALayerRemote::setDoubleSided):
(WebKit::PlatformCALayerRemote::setMasksToBounds):
(WebKit::PlatformCALayerRemote::setContentsRect):
(WebKit::PlatformCALayerRemote::setMinificationFilter):
(WebKit::PlatformCALayerRemote::setMagnificationFilter):
(WebKit::PlatformCALayerRemote::setBackgroundColor):
(WebKit::PlatformCALayerRemote::setBorderWidth):
(WebKit::PlatformCALayerRemote::setBorderColor):
(WebKit::PlatformCALayerRemote::setOpacity):
(WebKit::PlatformCALayerRemote::setFilters):
(WebKit::PlatformCALayerRemote::copyFiltersFrom):
(WebKit::PlatformCALayerRemote::setBlendMode):
(WebKit::PlatformCALayerRemote::setName):
(WebKit::PlatformCALayerRemote::setSpeed):
(WebKit::PlatformCALayerRemote::setTimeOffset):
(WebKit::PlatformCALayerRemote::setContentsScale):
(WebKit::PlatformCALayerRemote::setCornerRadius):
(WebKit::PlatformCALayerRemote::setAntialiasesEdges):
(WebKit::PlatformCALayerRemote::setShapeRoundedRect):
(WebKit::PlatformCALayerRemote::setShapePath):
(WebKit::PlatformCALayerRemote::setShapeWindRule):
(WebKit::PlatformCALayerRemote::updateCustomAppearance):
(WebKit::PlatformCALayerRemote::setEventRegion):
(WebKit::PlatformCALayerRemote::setScrollingNodeID):
(WebKit::PlatformCALayerRemote::setIsSeparated):
(WebKit::PlatformCALayerRemote::setIsSeparatedPortal):
(WebKit::PlatformCALayerRemote::setIsDescendentOfSeparatedPortal):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteScrollingCoordinator.mm:
(WebKit::RemoteScrollingCoordinator::scrollingStateInUIProcessChanged):

Canonical link: <a href="https://commits.webkit.org/259176@main">https://commits.webkit.org/259176@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e7a19115069a5f96382b9d1437daa560fcb93aca

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104208 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13298 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37123 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113419 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173710 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108143 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14373 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4213 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96435 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112477 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109977 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11072 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94134 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38736 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/107692 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92928 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25746 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80394 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6661 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27111 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6796 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3658 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12808 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46658 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8580 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3352 "Failed to push commit to Webkit repository") | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
```diff
@@ -58,7 +58,7 @@ class RemoteLayerBackingStore {
     ~RemoteLayerBackingStore();
 
-    enum class Type : uint8_t {
+    enum class Type : bool {
```
Why is `bool` better?

achristensen07 replied:
bool only has 2 valid states, so the decoder doesn't need any more metadata if an enum class uses bool as an underlying type.

smfr reviewed (approved):
Doing the enum class changes in a separate patch would make this easier to review, but this is OK.

achristensen07 committed:
Begin generating RemoteLayerTree serialization code
https://bugs.webkit.org/show_bug.cgi?id=250930
rdar://104505653

Reviewed by Simon Fraser.

* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.h:
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(typenames):
(remove_template_parameters):
(remove_alias_struct_or_class):
(alias_struct_or_class):
(generate_header):
(generate_serialized_type_info):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
Include some test expectation that should've been included in 259112@main
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.mm:
(WebKit::RemoteLayerBackingStoreCollection::prepareBackingStoresForDisplay):
(WebKit::RemoteLayerBackingStoreCollection::didFlushLayers):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in: Added.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm:
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToLayer):
(WebKit::RemoteLayerTreePropertyApplier::applyProperties):
(WebKit::RemoteLayerTreePropertyApplier::applyHierarchyUpdates):
(WebKit::RemoteLayerTreePropertyApplier::updateMask):
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToUIView):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(WebKit::RemoteLayerTreeTransaction::LayerProperties::encode const):
(WebKit::RemoteLayerTreeTransaction::LayerProperties::decode):
(WebKit::dumpChangedLayers):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStoreCollection.mm:
(WebKit::RemoteLayerWithRemoteRenderingBackingStoreCollection::prepareBackingStoresForDisplay):
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.cpp:
(ArgumentCoder<RequestedScrollData>::encode): Deleted.
(ArgumentCoder<RequestedScrollData>::decode): Deleted.
(ArgumentCoder<SnapOffset<float>>::encode): Deleted.
(ArgumentCoder<SnapOffset<float>>::decode): Deleted.
(ArgumentCoder<FloatScrollSnapOffsetsInfo>::encode): Deleted.
(ArgumentCoder<FloatScrollSnapOffsetsInfo>::decode): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingUIState.cpp:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingUIState.h:
(WebKit::RemoteScrollingUIState::changes const):
* Source/WebKit/Shared/RemoteLayerTree/SwapBuffersDisplayRequirement.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::updateLayerTree):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.cpp:
(WebKit::PlatformCALayerRemote::PlatformCALayerRemote):
(WebKit::PlatformCALayerRemote::recursiveBuildTransaction):
(WebKit::PlatformCALayerRemote::copyContentsFromLayer):
(WebKit::PlatformCALayerRemote::removeSublayer):
(WebKit::PlatformCALayerRemote::setSublayers):
(WebKit::PlatformCALayerRemote::removeAllSublayers):
(WebKit::PlatformCALayerRemote::appendSublayer):
(WebKit::PlatformCALayerRemote::insertSublayer):
(WebKit::PlatformCALayerRemote::replaceSublayer):
(WebKit::PlatformCALayerRemote::addAnimationForKey):
(WebKit::PlatformCALayerRemote::removeAnimationForKey):
(WebKit::PlatformCALayerRemote::setMask):
(WebKit::PlatformCALayerRemote::setClonedLayer):
(WebKit::PlatformCALayerRemote::setOpaque):
(WebKit::PlatformCALayerRemote::setBounds):
(WebKit::PlatformCALayerRemote::setPosition):
(WebKit::PlatformCALayerRemote::setAnchorPoint):
(WebKit::PlatformCALayerRemote::setTransform):
(WebKit::PlatformCALayerRemote::setSublayerTransform):
(WebKit::PlatformCALayerRemote::setHidden):
(WebKit::PlatformCALayerRemote::setContentsHidden):
(WebKit::PlatformCALayerRemote::setUserInteractionEnabled):
(WebKit::PlatformCALayerRemote::setBackingStoreAttached):
(WebKit::PlatformCALayerRemote::setGeometryFlipped):
(WebKit::PlatformCALayerRemote::setDoubleSided):
(WebKit::PlatformCALayerRemote::setMasksToBounds):
(WebKit::PlatformCALayerRemote::setContentsRect):
(WebKit::PlatformCALayerRemote::setMinificationFilter):
(WebKit::PlatformCALayerRemote::setMagnificationFilter):
(WebKit::PlatformCALayerRemote::setBackgroundColor):
(WebKit::PlatformCALayerRemote::setBorderWidth):
(WebKit::PlatformCALayerRemote::setBorderColor):
(WebKit::PlatformCALayerRemote::setOpacity):
(WebKit::PlatformCALayerRemote::setFilters):
(WebKit::PlatformCALayerRemote::copyFiltersFrom):
(WebKit::PlatformCALayerRemote::setBlendMode):
(WebKit::PlatformCALayerRemote::setName):
(WebKit::PlatformCALayerRemote::setSpeed):
(WebKit::PlatformCALayerRemote::setTimeOffset):
(WebKit::PlatformCALayerRemote::setContentsScale):
(WebKit::PlatformCALayerRemote::setCornerRadius):
(WebKit::PlatformCALayerRemote::setAntialiasesEdges):
(WebKit::PlatformCALayerRemote::setShapeRoundedRect):
(WebKit::PlatformCALayerRemote::setShapePath):
(WebKit::PlatformCALayerRemote::setShapeWindRule):
(WebKit::PlatformCALayerRemote::updateCustomAppearance):
(WebKit::PlatformCALayerRemote::setEventRegion):
(WebKit::PlatformCALayerRemote::setScrollingNodeID):
(WebKit::PlatformCALayerRemote::setIsSeparated):
(WebKit::PlatformCALayerRemote::setIsSeparatedPortal):
(WebKit::PlatformCALayerRemote::setIsDescendentOfSeparatedPortal):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteScrollingCoordinator.mm:
(WebKit::RemoteScrollingCoordinator::scrollingStateInUIProcessChanged):

Canonical link: https://commits.webkit.org/259176@main
